### PR TITLE
Javohir To'ychiyev | DB task

### PR DESCRIPTION
**Pull request by Java-407**

### CODE REVIEW

File: 01-db-main-task.sql [hunk 1]
```diff
@@ -0,0 +1,138 @@
+CREATE TABLE courses (
```
Comment (sabirjahn) [inline 01-db-main-task.sql]: write with - create table if not exists

File: 01-db-main-task.sql [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+SELECT first_name, last_name FROM students
+WHERE student_id NOT IN (SELECT student_id FROM enrollments);
+--3.3.1
+SELECT courses.course_name, COUNT(enrollments.student_id)FROM courses
+LEFT JOIN enrollments ON courses.course_id = enrollments.course_id 
+GROUP BY courses.course_name;
+--3.3.2
+SELECT courses.course_name FROM courses
+JOIN enrollments ON courses.course_id = enrollments.course_id
+GROUP BY courses.course_name ORDER BY COUNT(enrollments.student_id) DESC;
```
Comment (sabirjahn) [inline 01-db-main-task.sql]: wrong!!!
you have to 
- Find the course with the most students.
and please check others also

File: 01-db-main-task.sql [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+--3.4.2
+SELECT students.first_name, students.last_name FROM students
+JOIN enrollments ON students.student_id = enrollments.student_id
+JOIN courses ON enrollments.course_id = courses.course_id
+WHERE students.enrollment_year > 2015-01-01 and courses.course_name = 'Tarix';
+--3.5
+SELECT  courses.course_name FROM courses JOIN enrollments ON courses.course_id = enrollments.course_id
+WHERE enrollments.student_id = (SELECT enrollments.student_id FROM enrollments GROUP BY enrollments.student_id
+ORDER BY AVG(enrollments.grade) LIMIT 1);
```
Comment (sabirjahn) [inline 01-db-main-task.sql]: 3.5.1 is not finished!
also you have to
- Find students enrolled in more courses than the average number of courses per student

**Comment by Java-407** [comment]
Just check again

Чт, 28 нояб. 2024 г. в 21:46, Sobirjon ***@***.***>:

File: Javohir-To'ychiyev/01-db-main-task.sql [hunk 4]
```diff
@@ -0,0 +1,183 @@
+CREATE TABLE courses (
```
Comment (Nasa-133) [inline Javohir-To'ychiyev/01-db-main-task.sql]: write with - create table if not exists

**Review by sabirjahn — COMMENTED** [review]
write with - create table if not exists...

File: Javohir-To'ychiyev/01-db-main-task.sql [hunk 5]
```diff
@@ -1,16 +1,13 @@
-CREATE TABLE courses (
-    course_id SERIAL PRIMARY KEY,
-    course_name VARCHAR(100),
-    credit_hours INTEGER
+create database university;
+create table students(
```
Comment (sabirjahn) [inline Javohir-To'ychiyev/01-db-main-task.sql:2]: dear write - create table if not exists 
and that's it

**Review by sabirjahn — COMMENTED** [review]
3.3.2 is wrong
- Find the course with the most students.
be careful

**Pull request closed without merging**